### PR TITLE
[GH-248] add default settings

### PR DESCRIPTION
**Pull request by annaos**
- Add new configuration for default options
- Set default options to create poll dialog

Issue: https://github.com/matterpoll/matterpoll/issues/248

### CODE REVIEW

**Comment by annaos** [comment]
@hanzei Label `Work In Progress` can be removed. PR is ready for review

File: server/plugin/command_test.go [hunk 1]
```diff
@@ -71,24 +71,28 @@ func TestPluginExecuteCommand(t *testing.T) {
 				Name:        "setting-anonymous",
 				Type:        "bool",
 				Placeholder: "Don't show who voted for what when the poll ends",
+				Default:     "false",
```
Comment (kaakaa) [inline server/plugin/command_test.go:74]: Would it be possible for you to add a few tests for cases where some default settings is enabled?

**Review by kaakaa — COMMENTED** [review]
Nice work @annaos !
I left some comments, and would you please add brief descriptions about this setting in [README](https://github.com/matterpoll/matterpoll?tab=readme-ov-file#settings)?

File: webapp/src/components/admin_settings/default_settings.jsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    handleChange = (name, value) => {
+        this.settings[name] = value;
+        this.props.onChange(this.props.id, this.settings);
+        this.props.setSaveNeeded();
+    };
+
+    render() {
+        return (
+            <div>
+                <strong>{'Default settings'}</strong>
```
[the remark this answers is not in the document]
Comment (annaos) [reply webapp/src/components/admin_settings/default_settings.jsx]: Here are two screenshots for comparison. In my opinion, with the title we have too little space for the labels of the checkboxes.

with `showTitle`:
Bildschirmfoto vom 2024-04-08 10-01-47

without `showTitle`:
Bildschirmfoto vom 2024-04-08 10-02-58

File: plugin.json [hunk 3]
```diff
@@ -35,6 +35,12 @@
                 "type": "bool",
                 "help_text": "When true, Matterpoll will render poll posts with a rich UI. The rich UI is not available on the mobile app.",
                 "default": false
+            },
+            {
+                "key": "default_settings",
+                "display_name": "Default Settings",
+                "type": "custom",
+                "help_text": "Settings will be pre-selected in 'Create Poll' dialog."
```
[the remark this answers is not in the document]
Comment (annaos) [reply plugin.json]: done. Feel free to change my wording or make suggestions.

File: README.md [hunk 4]
```diff
@@ -24,6 +24,7 @@ You can configure Matterpoll from **System Console > Plugins > Matterpoll**.
 * **Experimental UI**: Enable new experimental UI for poll posts:
   - Change button color of voted answers
   - Hide poll management buttons (Add Option / Delete Poll / End Poll) from users who don't have permission
+* **Default Settings**: Choose settings, that will be pre-selected in 'Create Poll' dialog. Settings will be not applied to `poll` command.
```
Comment (kaakaa) [inline README.md:27]:
```suggestion
* **Default Settings**: Choose settings, that will be pre-selected in 'Create Poll' dialog. Settings will not be applied to `/poll` command.
```

File: plugin.json [hunk 5]
```diff
@@ -35,6 +35,11 @@
                 "type": "bool",
                 "help_text": "When true, Matterpoll will render poll posts with a rich UI. The rich UI is not available on the mobile app.",
                 "default": false
+            },
+            {
+                "key": "default_settings",
+                "type": "custom",
+                "help_text": "Settings will be pre-selected in 'Create Poll' dialog. Settings will be not applied to `\\poll` command."
```
Comment (kaakaa) [inline plugin.json:42]:
```suggestion
                "help_text": "Settings will be pre-selected in 'Create Poll' dialog. Settings will not be applied to `/poll` command."
```

File: webapp/src/components/admin_settings/default_settings.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (kaakaa) [reply webapp/src/components/admin_settings/default_settings.jsx]: Thank you for clarifying this @annaos . Certainly, UI with `showTitle=true` is not styled well, e.g.: the `Progress` label is not styled well with line breaks. However, I think that the disadvantage of not showing help_text on WebUI has more impact, as users cannot understand the effect of the setting change, so I'm inclined to use `showTitle=true`. What do you think about?

[the remark this answers is not in the document]
Comment (annaos) [reply webapp/src/components/admin_settings/default_settings.jsx]: I didn't notice that help_text was not displayed.  I will add it to the default_settings.jsx. Is that okay with you?

[the remark this answers is not in the document]
Comment (kaakaa) [reply webapp/src/components/admin_settings/default_settings.jsx]: Thank you for reworking, but, the current implementation looks odd to me in terms of positioning and margins of the help text. I have not seen this way of displaying help text in other settings, so as a user I feel it is not natural, and I don't want to have the code on the plugin side to tweak the position.
スクリーンショット 2024-04-18 21 23 43

I would still prefer to adopt the display by `showTitle`, but is it still difficult?

[the remark this answers is not in the document]
Comment (annaos) [reply webapp/src/components/admin_settings/default_settings.jsx]: done :blush:

File: plugin.json [hunk 6]
```diff
@@ -35,6 +35,12 @@
                 "type": "bool",
                 "help_text": "When true, Matterpoll will render poll posts with a rich UI. The rich UI is not available on the mobile app.",
                 "default": false
+            },
+            {
+                "key": "default_settings",
+                "display_name": "Default Settings",
+                "type": "custom",
+                "help_text": "Settings will be pre-selected in 'Create Poll' dialog. Settings will be not applied to `\\poll` command."
```
Comment (kaakaa) [inline plugin.json:43]:
```suggestion
                "help_text": "Settings will be pre-selected in 'Create Poll' dialog. Settings will not be applied to `/poll` command."
```

fix this wording again.

**Comment by kaakaa** [comment]
Thank you for your great working @annaos! I'll approve this PR after resolving my tiny suggestion. 👍

**Review by kaakaa — APPROVED** [review]

**Pull request merged**